crates/bevy_reflect/src/map.rs

```diff
@@ -381,6 +381,14 @@ pub struct MapIter<'a> {
     pub(crate) index: usize,
```

Comment:
To better align with `ListIter`, could we make these fields private? We'll also need to update the impls for `HashMap` (and we can do the same for `DynamicMap` just for consistency).